### ISSUE TITLE
Pull latest Classic McEliece code

### ISSUE THREAD

**Issue by jschanck** [PQClean/PQClean#392]
As discussed on our call, I'm going to script the inclusion of the Classic McEliece code.

A few questions:
* The latest code package for NIST is more recent and is not identical to the latest supercop code, but I would assume that supercop will be more reliably up-to-date in the long term. Are we OK with pulling from supercop?
* The upstream code has a number of 'static inline" functions defined in header files. Currently we move these to their own c files. Can I leave them in the headers instead? Moving them to separate c files is by far the biggest barrier to automatic updates.
* Do we want all four implementations of each parameter set? Liboqs currently only takes the "vec" and "avx" implementations. I wonder if there's real interest in "clean" (i.e. ref) and "sse" implementations.

**Comment by thomwiggers** [comment]
Sorry that I forgot to respond to your questions.

* The current code was also pulled from SUPERCOP iirc.
* We used to be a bit strict about this but maybe I should care less.
* PQClean intends to collect "clean reference implementations" so I would prefer to still have the `clean` implementation. `see` is maybe not that interesting indeed.


This might also solve these issues:

```
FAILED: src/kem/classic_mceliece/CMakeFiles/classic_mceliece_348864_avx.dir/pqclean_mceliece348864_avx/fft_tr.c.o
/usr/bin/cc  -Iinclude -I../src/kem/classic_mceliece/pqclean_mceliece348864_avx -I../src/common/pqclean_shims -fPIC -fvisibility=hidden -march=native -Werror -Wall -Wextra -Wpedantic -Wstrict-prototypes -Wshadow -Wformat=2 -Wfloat-equal -Wwrite-strings -O3 -fomit-frame-pointer -fdata-sections -ffunction-sections -Wl,--gc-sections -mavx2 -mpopcnt -Wno-language-extension-token -std=gnu11 -MD -MT src/kem/classic_mceliece/CMakeFiles/classic_mceliece_348864_avx.dir/pqclean_mceliece348864_avx/fft_tr.c.o -MF src/kem/classic_mceliece/CMakeFiles/classic_mceliece_348864_avx.dir/pqclean_mceliece348864_avx/fft_tr.c.o.d -o src/kem/classic_mceliece/CMakeFiles/classic_mceliece_348864_avx.dir/pqclean_mceliece348864_avx/fft_tr.c.o -c ../src/kem/classic_mceliece/pqclean_mceliece348864_avx/fft_tr.c
../src/kem/classic_mceliece/pqclean_mceliece348864_avx/fft_tr.c:351:47: error: argument 1 of type ‘vec128[12]’ {aka ‘__m128i[12]’} with mismatched bound [-Werror=array-parameter=]
  351 | void PQCLEAN_MCELIECE348864_AVX_fft_tr(vec128 out[GFBITS], vec256 in[][ GFBITS ]) {
      |                                        ~~~~~~~^~~~~~~~~~~
In file included from ../src/kem/classic_mceliece/pqclean_mceliece348864_avx/fft_tr.c:6:
../src/kem/classic_mceliece/pqclean_mceliece348864_avx/fft_tr.h:11:40: note: previously declared as ‘vec128 *’ {aka ‘__m128i *’}
   11 | void PQCLEAN_MCELIECE348864_AVX_fft_tr(vec128 * /*out*/, vec256  /*in*/[][ GFBITS ]);
      |                                        ^~~~~~~~
```

**Comment by thomwiggers** [comment]
Closed in favour of #459
